chainstate/src/interface/chainstate_interface_impl.rs

```diff
@@ -560,17 +560,18 @@ fn get_output_coin_amount(
         }
         TxOutput::StakePool(data) => Some(data.value()),
         TxOutput::ProduceBlockFromStake(_, pool_id) => {
```

Comment:
This isn't specifically for this PR, but this whole function (and the one calling it) seems to be playing with fire. There's no "output amount" for a coin. This is dangerous. If we want to allow such behavior, why didn't we allow something like `value()` in TxOutput?

Reply:
Agree, this function has to go. It's only used in mempool tests and was introduced to calculate fees, but is obsolete now.